DSharpPlus/Entities/Guild/DiscordRole.cs

```diff
@@ -57,6 +58,25 @@ public DiscordColor Color
         [JsonProperty("hoist", NullValueHandling = NullValueHandling.Ignore)]
         public bool IsHoisted { get; internal set; }
 
+        /// <summary>
+        /// The url for this role's icon, if set.
+        /// </summary>
+        public string IconUrl => this.IconHash != null ? $"https://cdn.discordapp.com/role-icons/{this.Id}/{this.IconHash}.png" : null;
+
+        /// <summary>
+        /// The hash of this role's icon, if any.
+        /// </summary>
+        [JsonProperty("icon")]
+        public string IconHash { get; internal set; }
+
+        /// <summary>
+        /// The emoji associated with this role's icon, if set.
+        /// </summary>
+        public DiscordEmoji Emoji => this._emoji != null ? DiscordEmoji.FromUnicode(this._emoji) : null;
```

Comment:
This will be a problem: Specified unicode entity is not a valid unicode emoji. (Parameter 'unicodeEntity')

See https://github.com/discord/discord-api-docs/pull/3878

Reply:
I had to do
```cs

        /// <summary>
        /// Gets the role unicode_emoji.
        /// </summary>
        [JsonProperty("unicode_emoji", NullValueHandling = NullValueHandling.Ignore)]
        internal string _unicodeEmojiString;

        /// <summary>
        /// Gets the unicode emoji.
        /// </summary>
        public DiscordEmoji UnicodeEmoji
            => this._unicodeEmojiString != null ? DiscordEmoji.FromName(this.Discord, $":{this._unicodeEmojiString}:", false) : null;
```

Reply:
Idfk what discord did in the api.
I get a wrong response then

Reply:
ah you use .ToString().
That's the thing.
Good